### ISSUE TITLE
initialState props argument missing props

### ISSUE THREAD

**Issue by pho3nixf1re** [dekujs/deku#179]
I have a source in propTypes that should contain a `user` object. The initialState prop however does not contain the `user` property. The object getting passed into the initialState instead has a `children` property that I'm not defining that has an array of length 0. My expectation from the below code snippet is that the `user` property should be in any of the props getting passed into various hooks.

Code...

```js
'use strict';

import { element } from 'deku';
import page from 'page';
import _ from 'lodash';
import {login} from 'app/stores/user';
import serializer from 'form-serialize';
import styles from './index.scss';
import Waiting from 'app/components/waiting';

const propTypes = {
    user: { type: 'object', source: 'user' }
};

function initialState(props) {
    console.log(props);
    return {
        status: _.isEmpty(props.user) ? 'input' : 'finished' // input, waiting, finished
    };
}

function handleSubmit(event, {}, updateState) {
    event.preventDefault();
    let data = serializer(event.target, { hash: true });

    updateState({ status: 'waiting' });

    login(data).then(() => {
        updateState({ status: 'finished' });
    }).catch(() => {
        updateState({ status: 'input' });
    });
}

function displayView(status) {
    switch (status) {
    case 'finished':
        page('/recipes');
        return <p>Already logged in.</p>;
    case 'waiting':
        return <Waiting />;
    default:
        return (
            <form onSubmit={handleSubmit} class={styles.form}>
                <input
                    type='email'
                    name='email'
                    placeholder='email'
                    required={true} />
                <input
                    type='password'
                    name='password'
                    placeholder='password'
                    required={true} />
                <button type='submit'>login</button>
            </form>
        );
    }
}

function render({state}) {
    let view = displayView(state.status);
    return (
        <div>
            {view}
        </div>
    );
}

export default { render, initialState, propTypes };

```

**Comment by pho3nixf1re** [comment]
So after working with this for a bit, I've tracked down the issue. The `user` property is getting set asynchronously, apparently _after_ the initialState is calculated. This causes the `user` property to be undefined and therefore false. So my observation is actually expected behavior. Instead I have added an `isLoggedIn` method to my user store to check synchronously the state of the user. Woops!

I could have probably put this check into the beforeMount hook and it would have had the correct behavior. But now I have a useful utility method on my user store.
